### DISCUSSION TITLE
FS 22 performance with CP & AD

### DISCUSSION THREAD

**Discussion by Misamangame** [Courseplay/Courseplay_FS22#2268] [General]
Hello everyone,

I have question for everyone out there who is using CP & AD.  I have almost 7 years old pc now.  I'm playing always on 4x maps and I'm big user of CP & AD driven vehicles I don't do any work myself I like to watch the job and manage.  So for me especially when harvests are running is not a problem to have 30-40 vehicles on CP & AD (I would like to have even more when my farm will grow bigger:) ).  And ofcourse it is affecting my FPS performance. General effect is depending on how many vehicles is running, pathfing etc. but game getting as low as 10FPS which sucks.  When I start my game I have 60FPS with no problem and with more and more vehicles running on workers it is gradualy descending.  
With new generation of CPUs and GPUs I will be upgrading my pc sometimes this year. And here is my question.  Is FS 22 as a game even able to use all that computing power????  For me it doesn't matter that I get more FPS with base game but when I'll be running CP & AD workers will drop in performance will be the same or less visible? Isn't it simply game engine limitation so even with newest pc there will be always performance problem ?  I would like to know if here is someone who is playing on bigger maps with many vehicles (20+) running on CP & AD how newer PC's are coping with that.

**Comment by Tensuko** [comment]
The game engine and LUA is the limit.
At some point, a better PC won't give your more FPS with so many vehicles running.

**Reply by Misamangame** [reply]
Hm this is exactly what I did not wanted to hear :)

**Reply by pvaiko** [reply]
As far as I can tell all scripts run in a single thread which is the main limiting factor. You can look at single thread benchmarks (like https://www.cpubenchmark.net/singleThread.html) to compare your current CPU performance with a newer one, that's a good starting point of what performance increase you can expect. 

I'm not quite sure about this, but a better video card may also help, giving the CPU more time.

**Reply by Misamangame** [reply]
@pvaiko thx for the chart it is great comparison.  Now I can see my pc is reaaallllyyyy old :). my GPU is actually newest thing in there but I will be upgrading whole package anyway.  Shame FS is not so much popular or graphic intense game to be tested by on regular basis with various CPU/GPU combos like many new not so good games by various YT channels

**Reply by Redster999** [reply]
I have a 4000$ PC. And stuttering and framerate drops is an issue with the game. Hopefully a solid update of the game could fix this. But knowing Giants history I doubt this would happen

**Reply by Misamangame** [reply]
Well this is about budget I want to make upgrade. But yes I do agree there will be no fix or patch from Giants. Like pvaiko mentioned if the game is not able to use hardware properly it is just bad.
